src/Database/PostgreSQL/PQTypes/Checks/Util.hs

```diff
@@ -218,13 +219,19 @@ checkOverlappingIndexesQuery =
     , "  SELECT contained"
     , "       , contains"
     , "  FROM indexdata2"
+    , " JOIN pg_class c ON (c.oid = indexdata2.indexrelid)"
     , -- The indexes are the same or the "other" is larger than us
       "  WHERE (colotherindex = colindex"
     , "      OR colotherindex LIKE colindex || '+%')"
+    , -- and this is not a local index
+      "    AND NOT c.relname ILIKE 'local_%'"
     , -- and we have the same predicate
       "    AND other_preddef IS NOT DISTINCT FROM preddef"
     , -- and either the other is unique (so better than us) or none of us is unique
-      "    AND (other_indisunique"
-    , "      OR (NOT other_indisunique"
-    , "      AND NOT indisunique));"
+      "    AND (NOT indisunique)"
+    , "    OR ("
+    , "             indisunique"
+    , "         AND other_indisunique"
+    , "         AND colindex = colotherindex"
+    , ");"
```

Comment:
@marco44 : OK for you ? (Could you please confirm l. 227 and l. 205 seem ok ?)

Reply:
Seems ok to me